src/hotspot/share/prims/jni.cpp

```diff
@@ -3633,7 +3633,13 @@ static jint JNI_CreateJavaVM_inner(JavaVM **vm, void **penv, void *args) {
         HandleMark hm(THREAD);
         vm_exit_during_initialization(Handle(THREAD, PENDING_EXCEPTION));
       }
+      // Unclear if we can actually reach here as all error returns should have
+      // set an exception pending. But it is okay to fall-through to the general
+      // termination code below.
```

Comment:
If it were true that all error returns here had an exception pending then we wouldn't need the conditional
smr_deletes in create_vm.  And if we return without an exception then we'll also not have a current thread
and the JavaThread::current call will return null (or assert in a debug build), leading to failures.  That is,
this change doesn't seem to have addressed the issue described in the bug at all.

Reply:
I think you are right - I tried to make too many changes. The original code just didn't delete the main-thread - nice and simple but "leaks". With Coleen's suggested change we only keep the main thread if there is an exception pending, but that then opens up the possibility that we reach this section of code with no current thread again.